tests/test_minimal/test_metadata_schema.py

```diff
@@ -0,0 +1,201 @@
+                    dff=dict(name="RoiResponseSeriesChan1Plane0", description="Array of df/F traces."),
+                ),
+                PlaneSegmentationChan1Plane1=dict(
+                    dff=dict(name="RoiResponseSeriesChan1Plane0", description="Array of df/F traces."),
+                ),
+            ),
+            SegmentationImages=dict(
+                name="SegmentationImages",
+                PlaneSegmentationChan1Plane0=dict(
+                    raw=dict(name="PlaneSegmentationChan1Plane0", description="raw segmentation image"),
+                    neuropil=dict(name="PlaneSegmentationChan1Plane0", description="neuropil segmentation image"),
+                ),
+            ),
+        ),
+    )
+
+    validate_metadata(metadata=metadata, schema=metadata_schema)
+
+
+device_error = "'Devices' is a required property"
+invalid_plane_name = "plane_segmentation_chan1_plane0"
+name_error = "'name' is a required property"
+description_error = "'description' is a required property"
+plane_name_error = f"'{invalid_plane_name}' does not match any of the regexes"
+not_enough_properties_error = "does not have enough properties"
+
+
+def test_invalid_ophys_metadata():
+    metadata_schema = load_dict_from_file(
+        Path(__file__).parent.parent.parent / "src" / "neuroconv" / "schemas" / "metadata_schema.json"
+    )
+
+    def create_valid_plane_name(name: str, number=0):
+        return f'{"".join(map(lambda str: str.capitalize(), name.split("_")))[:-1]}{str(number)}'
+
+    metadata = dict(
+        NWBFile=dict(
+            session_start_time="2020-01-01T00:00:00",
+            session_description="Auto-generated by neuroconv",
+            identifier="1234",
+        ),
+        Ophys=dict(
+            Fluorescence={
+                "name": "Fluorescence",
+                invalid_plane_name: dict(),  # Unchecked
+                create_valid_plane_name(invalid_plane_name): dict(),
+                create_valid_plane_name(invalid_plane_name, 1): dict(
+                    raw=dict(),
+                ),
+            },
+            DFOverF={
+                "name": "DfOverF",
+                invalid_plane_name: dict(),  # Unchecked
+                create_valid_plane_name(invalid_plane_name): dict(),
+                create_valid_plane_name(invalid_plane_name, 1): dict(
+                    raw=dict(),
+                ),
+            },
+            SegmentationImages={
+                "name": "SegmentationImages",
+                invalid_plane_name: dict(),  # Unchecked
+                create_valid_plane_name(invalid_plane_name): dict(),
+                create_valid_plane_name(invalid_plane_name, 1): dict(
+                    raw=dict(),
+                ),
+            },
+        ),
+    )
+
+    validator = jsonschema.Draft7Validator(metadata_schema)
+
+    errors = list(map(lambda e: str(e).split("\n")[0], validator.iter_errors(metadata)))
+
+    iterable = iter(errors)
```

Comment:
```suggestion
```

Reply:
Moved to a method that searches the error list for matching errors instead of directly checking at a certain index.